### DISCUSSION TITLE
Support @Check decorator (creates constraints) for all DB drivers

### DISCUSSION THREAD

**Discussion by maxime-dupuis** [mikro-orm/mikro-orm#5285] [General]
**Is your feature request related to a problem? Please describe.**

I'd like to have data validation in the database layer rather than application layer. This way all my applications will benefit from it.
([Similar issue](https://github.com/mikro-orm/mikro-orm/issues/1711) was closed after it was implemented for the Postgres driver only )

The [@Check documentation](https://mikro-orm.io/docs/decorators#check) says that: 

> Check constraints are currently supported only in postgres driver.

**Describe the solution you'd like**

`@Check` decorator works for all database drivers

**Describe alternatives you've considered**

[Class validators](https://github.com/typestack/class-validator?tab=readme-ov-file#custom-validation-classes) at the application level

**Comment by B4nan** [comment]
It already works in most drivers, including mysql 8 and mariadb. Only sqlite currently does not support those - or better say, schema diffing does not infer those from the database, so diffing the database will always try to create them even if they are present.

**Reply by B4nan** [reply]
Looking at the sqlite support now, and it might be rather easy, since we already have the schema inference implemented for enums (that use check constraints), so it should be mostly about moving the code around a bit.

**Reply by B4nan** [reply]
Ok, now I remember why this isn't implemented, sqlite does not allow adding check constraints to existing tables.

**Reply by maxime-dupuis** [reply]
Thanks for the info! The doc lead me to believe it was Postgres only, but I'll start using checks for MySQL :)

**Reply by B4nan** [reply]
Fixed here 78dfbba104c4e75cfa5204142ec27194149cbc7f
